File: apps/actor-ios/ActorClient/CocoaDispatcherProvider.swift
//
//  Copyright (c) 2015 Actor LLC. <https://actor.im>
//

import Foundation
class CocoaDispatcherProvider: NSObject, AMDispatcherProvider {

    func dispatchWithRunnable(runnable: JavaLangRunnable!) {
        dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_BACKGROUND, 0), {
            runnable.run()
        });
    }
}

File: actor-sdk/sdk-core/runtime/cocoa/src/main/java/im/actor/runtime/cocoa/CocoaDispatcherProvider.java
package im.actor.runtime.cocoa;

import im.actor.runtime.DispatcherRuntime;

public class CocoaDispatcherProvider implements DispatcherRuntime {

    @Override
    public native void dispatch(Runnable runnable)/*-[
        dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_BACKGROUND, 0), ^{
            [runnable run];
        });
    ]-*/;
}
